### DISCUSSION TITLE
Unable to pop up Alert in SPM's Preview

### DISCUSSION THREAD

**Discussion by fatbobman** [pointfreeco/swift-composable-architecture#1614] [General]
I was previewing the Feature's view directly in SPM and everything was working fine until I tried to pop up the Alert and the preview crashed.

However, when I test this by importing the Package in Xcode Project, the preview is perfectly fine.

I'm currently developing under Xcode 14, and this issue is also present in the new 14.1 version.

Since the preview crashes without a hint, there is no way to find out what the problem is.
I wonder if anyone else has encountered a similar problem?

<img width="349" alt="image" src="https://user-images.githubusercontent.com/55673881/199640332-0e62888c-bc14-4c41-8180-98ff328b1650.png">

**Comment by mbrandonw** [comment]
Hi @fatbobman, sorry I have not seen this behavior before and the [alerts case study](https://github.com/pointfreeco/swift-composable-architecture/blob/3bfbc7f9b89dccb6e7bf845dff80cb3cf7d23ba9/Examples/CaseStudies/SwiftUICaseStudies/01-GettingStarted-AlertsAndConfirmationDialogs.swift) seems to work fine. Do you have a project you can share that reproduces?

**Comment by tgrapperon** [comment]
I can furthermore confirm that the alert case study also works from a pure SwiftPM context.
[TCAAlert.zip](https://github.com/pointfreeco/swift-composable-architecture/files/9931846/TCAAlert.zip)
You can maybe try to spin an app that launches the preview, which should make debugging easier:
```swift
@main
struct PreviewApp: App {
  var body: some Scene {
    WindowGroup {
      MyFeatureView_Previews.previews
    }
  }
}
```

**Comment by fatbobman** [comment]
I found the reason why Preview crashed due to pop-up alert. It's not a TCA problem, it seems to be an XCode bug at the moment
In the presence of NavigationStack + toolbar in the view, as long as you try to pop up alert , the preview will inevitably crash
alert doesn't crash if toolbar is commented out

```swift
public struct AlertCrashPreview: PreviewProvider {
    public static var previews: some View {
        AlertCrashView()
    }
}

struct AlertCrashView: View {
    @State var show = false
    var body: some View {
        NavigationStack {
            VStack {
                Button("show") { show.toggle() }
            }
            .alert(isPresented: $show) {
                Alert(title: Text("abc"))
            }
            // alert doesn't crash if toolbar is commented out
            .toolbar {
                ToolbarItem {
                    Button("Del") {}
                }
            }
        }
    }
}
```

![iShot_2022-11-04_10 14 37 2022-11-04 10_15_33](https://user-images.githubusercontent.com/55673881/199870702-bceba990-550f-4a43-ab43-58f461fb7e2c.gif)
